syntax = "proto3";

package intellgraph;

message VertexParameter {
  // Required
  int32 id = 1;

  // Required
  enum Type {
    INPUT = 0;
    HIDDEN = 1;
    OUTPUT = 2;
  }
  Type type = 2;

  // Required
  string operation = 3;

  // Required
  int32 dims = 4;
}
